meshnets/process.py

```diff
@@ -14,6 +15,8 @@
 flags.DEFINE_string('data_dir', os.path.join('data', 'dataset'),
                     'Path to the folder containing the mesh files.')
 
+flags.DEFINE_float('tolerance', 1.5, 'Tolerance for the mesh sampling.')
```

Comment:
To be honest I am still trying to fully understand it myself.

From the documentation we have that:

![image](https://github.com/inductiva/meshnets/assets/33664950/25c475e6-5c5d-46c9-8ee7-b10e15cef1d9)


However, `source` and `input` are not variables of the method:

![image](https://github.com/inductiva/meshnets/assets/33664950/6b740bec-5ef3-4634-bd1d-27ec975d1a90)

I also tried to take a look at the source code where they use `source` for two different things:

![image](https://github.com/inductiva/meshnets/assets/33664950/98a431f7-75e7-4d85-9fda-bbdfe48c6192)

So it seems that the `input` is the mesh to which we want to transfer values.

Therefore I think that this tolerance is a value to decide whether a point in the mesh shape we want to keep is in a cell of the other mesh. This will depend on the specific location algorithm that they use, which is not specified in the docs.

Moreover, interpolation in pyvista in hidden behing vtk:

![image](https://github.com/inductiva/meshnets/assets/33664950/22a970a0-7482-4eaf-93f7-ebf4d7b08f66)

What I did was test this with several different tolerance and look at the results:

tolerance 1:

OpenFOAM mesh             |  Interpolation to original mesh
:-------------------------:|:-------------------------:
![image](https://github.com/inductiva/meshnets/assets/33664950/ae8cfd7e-a268-455a-b3ec-27c53260c527) | ![image](https://github.com/inductiva/meshnets/assets/33664950/97cdf7ef-f926-4667-80be-440b2ea7426c)


Tolerance 0.001:
OpenFOAM mesh             |  Interpolation to original mesh
:-------------------------:|:-------------------------:
 ![image](https://github.com/inductiva/meshnets/assets/33664950/0fe0b969-bc65-4438-b039-97aee6e6c59e) | ![image](https://github.com/inductiva/meshnets/assets/33664950/0144176e-bbe8-486d-b12d-33785f173991)


Tolerance 0.00001:
OpenFOAM mesh             |  Interpolation to original mesh
:-------------------------:|:-------------------------:
![image](https://github.com/inductiva/meshnets/assets/33664950/7cf5c24a-914a-4d69-b3ed-e8e5ae5310f8)| ![image](https://github.com/inductiva/meshnets/assets/33664950/34d5fbc4-7f94-4298-b584-61f0f82e74dd)


Tolerance 1000:
OpenFOAM mesh             |  Interpolation to original mesh
:-------------------------:|:-------------------------: 
![image](https://github.com/inductiva/meshnets/assets/33664950/26e6fe85-d537-43e8-91d9-b7d8b14c2771)| ![image](https://github.com/inductiva/meshnets/assets/33664950/633cb877-e3ff-409d-bb23-b405c842dbc9)

So it seems that when we decrease the tolerance the interpolated values start moving towards an average of all the poinst in the mesh.

And that when we increase the tolerance the values are actually more fine grained.

But I need to dig deeper into this to be able to give of a satisfactory answer as to how this value affects the results. At the moment, the only justification for the chosen tolerance is that the results are visually aligned.

However, we can later have a sanity check tool that, for both meshes computes a histogram on the sistribution of the numerical values and see if they align for the OpenFOAM mesh and the interpolated mesh.

Maybe @IvanPombo can provide some light on this topic

Reply:
This tolerance is a value that needs to be manipulated according to the density of both meshes - if one is finer than the other. 

The sample method of pyvista states that the `source` is the mesh of object where we want to insert the values of the pressure field, and the `input` is actually the source we are sampling from, i.e., the mesh coming from OpenFOAM. Notice, that the mesh from OpenFOAM is mostly squares and in the other it's triangles. To make a square one usually needs at least two triangles. The tolerance is here to validate if the points of the mesh of the object are inside the cell of the openFOAM mesh with a certain margin for error. Notice that, a point of the mesh can be out of the cell, but most of the triangle be inside the cell, hence this triangle should actually take the value of that square and not one on the left... 

This sample method actually is interpolating based on close values, so this tolerance allows to consider close cells.

Reply:
Thanks a lot Augusto for going much deeper on this. But Ivan, the question remains: what does 1.5 mean?

Reply:
To be honest, I am not 100% sure about this, but it seems to be a relative value with regards to the edge length of the object mesh we want to set values in. I have searched up to the VTK repo, which is the basis for these algorithms in pyvista, but even there the documentation states the same as the one here - only trying to understand a bit of the code it seems that this is explained.